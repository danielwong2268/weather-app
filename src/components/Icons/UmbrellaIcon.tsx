import React from 'react';
import scaleStyle from 'src/styles/scaleStyle';


const UmbrellaIcon = () => (
  <div className={scaleStyle}>
    <svg className="scale" version="1.1" id="Capa_1" x="0px" y="0px" viewBox="0 0 60 60">
      <path style={{
          fill:'none',
          stroke: '#424A60',
          strokeWidth:2,
          strokeLinecap: 'round',
          strokeMiterlimit:10}}
          d="M30,1v54c0,2.209-1.791,4-4,4
        s-4-1.791-4-4"/>
      <path style={{fill:'#7383BF'}} d="M30,28c3.316-4,8.684-4,12,0c3.316-4,8.684-4,12,0C54,14.745,43.255,4,30,4S6,14.745,6,28
        c3.316-4,8.684-4,12,0C21.316,24,26.684,24,30,28z"/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
      <g/>
    </svg>
  </div>
);

export default UmbrellaIcon;