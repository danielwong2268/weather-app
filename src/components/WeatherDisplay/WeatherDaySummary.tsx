import React from 'react';
import styled, { css } from 'react-emotion';
import { GRAY_1 } from 'src/constants/colors';
import IconTooltip from '../IconTooltip/IconTooltip';
import UmbrellaIcon from '../Icons/UmbrellaIcon';
import JacketIcon from '../Icons/JacketIcon';
import { WeatherDaySummary } from 'src/types/weatherData';

const Container = styled('div')`
  display: flex;
  margin: 5px 10px;
  align-items: center;
  border: 1px solid ${GRAY_1};
  border-radius: 3px;
  position: relative;
`;

const Icons = styled('div')`
  position: absolute;
  bottom: 0px;
  right: 5px;
`;

interface WeatherSummaryDisplayProps {
  summary: WeatherDaySummary;
  showUmbrella: boolean;
  showJacket: boolean;
}

const WeatherDay = ({
  summary,
  showUmbrella,
  showJacket
}: WeatherSummaryDisplayProps) => (
  <Container>
    <img
      className={css`width:50px; height: 50px;`}
      src={`http://openweathermap.org/img/w/${summary.icon}d.png`}
      alt="weather icon"
    />
    <div className={css`margin-left: 10px; text-align: left;`}>
      <div className={css`font-weight: bold;`}>{summary.day}</div>
      <div>{summary.minTemp} - {summary.maxTemp}Â°F</div>
      <div>{summary.overallDesc}</div>
    </div>
    <Icons>
      { showUmbrella && <IconTooltip text="Sell umbrellas on this day!"><UmbrellaIcon /></IconTooltip> }
      { showJacket && <IconTooltip text="Sell jackets on this day!"><JacketIcon /></IconTooltip> }
    </Icons>
  </Container>
);

export default WeatherDay;