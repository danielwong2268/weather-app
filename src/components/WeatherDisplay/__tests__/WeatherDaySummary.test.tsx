import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import WeatherDaySummary from '../WeatherDaySummary';
import { shallow } from 'enzyme';
import JacketIcon from 'src/components/Icons/JacketIcon';
import UmbrellaIcon from 'src/components/Icons/UmbrellaIcon';

const testSummary = {
    maxTemp: 80,
    minTemp: 70,
    day: '10/21',
    isRaining: true,
    icon: '01',
    overallDesc: 'rainy'
}

describe('<WeatherDaySummary />', () => {
  it('Renders with summary data and icons', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<WeatherDaySummary summary={testSummary} showJacket showUmbrella />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Does not render jacket icon if showJacket is false', () => {
    const wrapper = shallow(<WeatherDaySummary summary={testSummary} showJacket={false} showUmbrella />);

    expect(wrapper.find(JacketIcon).exists()).toBeFalsy();
  });

  it('Does not render umbrella icon if showUmbrella is false', () => {
    const wrapper = shallow(<WeatherDaySummary summary={testSummary} showJacket showUmbrella={false} />);

    expect(wrapper.find(UmbrellaIcon).exists()).toBeFalsy();
  });
});