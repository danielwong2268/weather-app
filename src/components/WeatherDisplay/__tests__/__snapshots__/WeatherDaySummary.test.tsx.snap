// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDaySummary /> Renders with summary data and icons 1`] = `
<Styled(div)>
  <img
    alt="weather icon"
    className="css-da14pc"
    src="http://openweathermap.org/img/w/01d.png"
  />
  <div
    className="css-135wvy2"
  >
    <div
      className="css-1b1bivf"
    >
      10/21
    </div>
    <div>
      70
       - 
      80
      Â°F
    </div>
    <div>
      rainy
    </div>
  </div>
  <Styled(div)>
    <IconTooltip
      text="Sell umbrellas on this day!"
    >
      <UmbrellaIcon />
    </IconTooltip>
    <IconTooltip
      text="Sell jackets on this day!"
    >
      <JacketIcon />
    </IconTooltip>
  </Styled(div)>
</Styled(div)>
`;
